src/components/YourCompany/FolioNumber.vue

```diff
@@ -1,89 +1,262 @@
+                <v-list-item
+                  class="v-list-item change-button"
+                  @click="isEditing = true; dropdown = false"
+                >
+                  <v-list-item-subtitle>
+                    <v-icon small color="primary">mdi-pencil</v-icon>
+                    <span class="drop-down-action ml-1">Change</span>
+                  </v-list-item-subtitle>
+                </v-list-item>
+              </v-list>
+            </v-menu>
+          </span>
+        </div>
       </v-col>
     </v-row>
+
+    <!-- Edit Folio Number -->
+    <v-form v-else id="edit-folio-number" v-model="formValid" @submit.prevent="onFormSubmit()">
+      <!-- Line 1 -->
+      <v-row no-gutters>
+        <v-col cols="3">
+          <label :class="{'error-text': invalidSection}">
+            <strong>Business Folio or<br>Reference Number</strong>
+          </label>
+        </v-col>
+        <v-col cols="9" />
+      </v-row>
+
+      <!-- Line 2 -->
+      <v-row no-gutters>
+        <v-col cols="3" />
+        <v-col cols="9" class="my-4">
+          <p>This is the Folio or Reference Number for this business for your own tracking purposes.
+            There is no fee to change this number. Any changes made will be applied immediately.</p>
+        </v-col>
+      </v-row>
+
+      <!-- Line 3 -->
+      <v-row no-gutters>
+        <v-col cols="3" />
+        <v-col cols="9">
+            <v-text-field
+              filled persistent-hint
+              id="folio-number-input"
+              autocomplete="chrome-off"
+              label="Folio or Reference Number (Optional)"
+              v-model="folioNumber"
+              :name="Math.random()"
+              :rules="rules"
+            />
+        </v-col>
+      </v-row>
+
+      <!-- Form Actions -->
+      <v-row no-gutters>
+        <v-col>
+          <div class="form-actions">
+            <v-btn
+              large color="primary"
+              class="save-button"
+              type="submit"
+              value="Submit"
+            >
+              <span>Save</span>
+            </v-btn>
+            <v-btn
+              id="cancel-btn"
+              large outlined color="primary"
+              @click="onCancelClicked()"
+            >
+              <span>Cancel</span>
+            </v-btn>
+          </div>
+        </v-col>
+      </v-row>
+    </v-form>
   </div>
 </template>
 
 <script lang="ts">
-// Libraries
 import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
 
 @Component({})
 export default class FolioNumber extends Vue {
   // Props
-  @Prop({ default: null })
-  readonly initialValue: string
+  @Prop({ default: null }) readonly initialValue: string
+  @Prop({ default: null }) readonly originalValue: string
+  @Prop({ default: false }) readonly disableActions: boolean
+  @Prop() readonly editLabel: string
+  @Prop() readonly editedLabel!: string
+  @Prop({ default: false }) readonly invalidSection!: boolean
```

Comment:
As this will be a shared component (future PR), this component does not interface with the store or services. Only props and events.

It is interesting to note that this component (like Business Contact Info), manages its own "edit state" and tells its parent about it (instead of being told by the parent).